nutkit/protocol/responses.py

```diff
@@ -196,19 +196,105 @@ def __eq__(self, _):
             else:
                 data["serverInfo"]["address"] = AnyAddress()
         # ----------------------------------------------------------------------
+        # TODO: remove block when all drivers support the fields
+        # ----------------------------------------------------------------------
+        from tests.shared import get_driver_name
+        if get_driver_name() in ["javascript"]:
+            # already sends counters but the wrong format and not all fields
+            if "_stats" in data["counters"]:
+                del data["counters"]
+            else:
+                import warnings
+                warnings.warn(
+                    "Backend supports well-formatted counter. "
+                    "Remove the backwards compatibility check!"
+                )
+        if get_driver_name() in ["python", "java", "javascript", "go",
+                                 "dotnet"]:
+            if "counters" in data:
+                import warnings
+                warnings.warn(
+                    "Backend supports counters field in Summary. "
+                    "Remove the backwards compatibility check!"
+                )
+            else:
+                data["counters"] = {
+                    "constraintsAdded": None,
+                    "constraintsRemoved": None,
+                    "containsSystemUpdates": None,
+                    "containsUpdates": None,
+                    "indexesAdded": None,
+                    "indexesRemoved": None,
+                    "labelsAdded": None,
+                    "labelsRemoved": None,
+                    "nodesCreated": None,
+                    "nodesDeleted": None,
+                    "propertiesSet": None,
+                    "relationshipsCreated": None,
+                    "relationshipsDeleted": None,
+                    "systemUpdates": None
+                }
+            for field in (
+                "database", "notifications", "plan", "profile",
+                "parameters", "query", "queryType", "resultAvailableAfter",
+                "resultConsumedAfter"
+            ):
+                if field in data:
+                    import warnings
+                    warnings.warn(
+                        "Backend supports %s field in Summary. "
+                        "Remove the backwards compatibility check!" % field
+                    )
+                else:
+                    data[field] = None
+        # ----------------------------------------------------------------------
         self.server_info = ServerInfo(**data["serverInfo"])
+        self.counters = Counters(**data["counters"])
+        self.database = data["database"]
+        self.notifications = data["notifications"]
+        self.plan = data["plan"]
+        self.profile = data["profile"]
+        self.parameters = data["parameters"]
```

Comment:
I haven't actually seen this in java implementation.....

Reply:
I saw this in the docs `ResultSummary.query().parameters()`. Note that the query parameters are nested inside a query object for some drivers, while others expose them at summary level.

Reply:
Actually, after checking, I found that only Python goes without the nesting. Maybe something to consider for 5.0. Do you think I should adjust the testkit protocol here to reflect how most drivers do it, @injectives?

Reply:
I believe we have 2 options:
1. Update to work as a majority for now.
2. Discuss unification and work from there. We could update drivers and retain the previous methods in deprecated state until 5.0.
Just pick which option sounds better.

Reply:
I chose both. 1. and 2. We can't unify the surface until 5.0 either way.